### PR TITLE
feat: implement email verification flow

### PR DESCRIPTION
**Pull request by dextertanyj**
Sample email:
image

### CODE REVIEW

File: frontend/src/pages/VerificationPage.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import { ApiResponseError } from "src/services/ApiService";
+
+export const VerificationPage = () => {
+  const { search } = useLocation();
+  const params = new URLSearchParams(search);
+  const userId = params.get("userId");
+  const code = params.get("code");
+  const { enqueueSnackbar } = useSnackbar();
+  const navigate = useNavigate();
+
```
Comment (junlong4321) [inline frontend/src/pages/VerificationPage.tsx:16]: Omit spacing

File: frontend/src/pages/VerificationPage.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  useEffect(() => {
+    if (userId === null || code === null) {
+      enqueueSnackbar("Something went wrong", {
+        variant: "error",
+        // Prevent double snackbar during development caused by
+        // React.StrictMode's double invocation
+        key: "Generic Error",
+        preventDuplicate: true,
+      });
+      navigate("/");
```
Comment (junlong4321) [inline frontend/src/pages/VerificationPage.tsx:30]: Change navigation "/" to a button which resends email

File: frontend/src/pages/VerificationPage.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        // TODO: Redirect to request resend verification email page.
+        navigate("/");
+      }
+    };
+    verify();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return (
+    <Container
```
Comment (junlong4321) [inline frontend/src/pages/VerificationPage.tsx:60]: Implement a component Center which has the below properties

**Review by junlong4321 — APPROVED** [review]

File: frontend/src/pages/VerificationPage.tsx (hunk 2 above)
Comment (dextertanyj) [reply frontend/src/pages/VerificationPage.tsx:30]: Will resolve in #107.

**Pull request merged**